contracts/badger-sett/strategies/BaseStrategyV2.sol

```diff
@@ -0,0 +1,331 @@
+    }
+
+    // NOTE: must exclude any tokens used in the yield
+    // Controller role - withdraw should return to Controller
+    function withdrawOther(address _asset) external virtual whenNotPaused returns (uint256 balance) {
+        _onlyController();
+        _onlyNotProtectedTokens(_asset);
+
+        balance = IERC20Upgradeable(_asset).balanceOf(address(this));
+        IERC20Upgradeable(_asset).safeTransfer(controller, balance);
+    }
+
+    /// ===== Permissioned Actions: Authoized Contract Pausers =====
+
+    function pause() external {
+        _onlyAuthorizedPausers();
+        _pause();
+    }
+
+    function unpause() external {
+        _onlyAuthorizedPausers();
+        _unpause();
+    }
+
+    /// ===== Internal Helper Functions =====
+
+    /// @notice If withdrawal fee is active, take the appropriate amount from the given value and transfer to rewards recipient
+    /// @return The withdrawal fee that was taken
+    function _processWithdrawalFee(uint256 _amount) internal returns (uint256) {
+        if (withdrawalFee == 0) {
+            return 0;
+        }
+
+        uint256 fee = _amount.mul(withdrawalFee).div(MAX_FEE);
+        IERC20Upgradeable(want).safeTransfer(IController(controller).rewards(), fee);
+        return fee;
+    }
+
+    /// @dev Helper function to process an arbitrary fee
+    /// @dev If the fee is active, transfers a given portion in basis points of the specified value to the recipient
+    /// @return The fee that was taken
+    function _processFee(
+        address token,
+        uint256 amount,
+        uint256 feeBps,
+        address recipient
+    ) internal returns (uint256) {
+        if (feeBps == 0) {
+            return 0;
+        }
+        uint256 fee = amount.mul(feeBps).div(MAX_FEE);
+        IERC20Upgradeable(token).safeTransfer(recipient, fee);
+        return fee;
+    }
+
+    /// @dev Reset approval and approve exact amount
+    function _safeApproveHelper(
+        address token,
+        address recipient,
+        uint256 amount
+    ) internal {
+        IERC20Upgradeable(token).safeApprove(recipient, 0);
+        IERC20Upgradeable(token).safeApprove(recipient, amount);
+    }
+
+    function _transferToVault(uint256 _amount) internal {
+        address _vault = IController(controller).vaults(address(want));
+        require(_vault != address(0), "!vault"); // additional protection so we don't burn the funds
+        IERC20Upgradeable(want).safeTransfer(_vault, _amount);
+    }
+
+    /// @notice Swap specified balance of given token on Uniswap with given path
+    function _swap_uniswap(
+        address startToken,
+        uint256 balance,
+        address[] memory path
+    ) internal {
+        _safeApproveHelper(startToken, uniswap, balance);
+        IUniswapRouterV2(uniswap).swapExactTokensForTokens(balance, 0, path, address(this), now);
+    }
+
+    /// @notice Swap specified balance of given token on Uniswap with given path
+    function _swap_sushiswap(
```

Comment:
@andy8052 It looks you're trying to add sushiswap support in the base strategy.

Might be worth looking into making the DEX configurable?  And just pass in sushiswap for your strategy and uniswap for others? 

Forking the base strategy class to support every DEX and having heterogenous interfaces across strategies is going to make the test harness more complicated and both will be hard to maintain...

Reply:
I agree with moving the swap logic out of the base strategy. Perhaps simply another StrategyBaseSwapper that inherits.
I agree with configurable DEX, but I think the hardcoded is good for these particular strategies